pygmt/mathops.py

```diff
@@ -0,0 +1,59 @@
+"""
+GMT modules for Mathematical operations on tables or grids
+"""
+from .clib import Session
+from .exceptions import GMTInvalidInput
+from .helpers import build_arg_string, fmt_docstring, kwargs_to_strings, use_alias
+
+
+@fmt_docstring
+@use_alias(C="cmap", T="series", G="truncate", H="output", I="reverse")
+@kwargs_to_strings(T="sequence", G="sequence")
+def makecpt(**kwargs):
+    """
+    Creates a static color palette table (CPT).
+
+    Full option list at :gmt-docs:`makecpt.html`
+
+    Parameters
+    ----------
+    cmap (C) : str
+        Selects the master color palette table (CPT) to use in the interpolation.
+        Full list of built-in color palette tables can be found at
+        :gmt-docs:`GMT_Docs.html#built-in-color-palette-tables-cpt`.
+
+    series (T) : list or str
+        ``[min/max/inc[+b|l|n]|file|list]``.
+        Defines the range of the new CPT by giving the lowest and highest z-value (and
+        optionally an interval). If this is not given, the existing range in the master
+        CPT will be used intact.
+
+    truncate (G) : list or str
+        ``zlo/zhi``.
+        Truncate the incoming CPT so that the lowest and highest z-levels are to zlo and
+        zhi. If one of these equal NaN then we leave that end of the CPT alone. The
+        truncation takes place before any resampling. See also
+        :gmt-docs:`cookbook/features.html#manipulating-cpts`.
+
+    output (H) : str
+        Optional. The file name with extension .cpt to store the generated CPT file.
+        If not given or False (default), saves the CPT as the session current CPT.
+
+    reverse (I) : str
+        Set this to True or c [Default] to reverse the sense of color progression in the
+        master CPT. Set this to z to reverse the sign of z-values in the color table.
+        Note that this change of z-direction happens before -G and -T values are used so
+        the latter must be compatible with the changed z-range. See also
+        :gmt-docs:`cookbook/features.html#manipulating-cpts`.
+
+    {aliases}
+    """
+    with Session() as lib:
+        if "H" not in kwargs.keys():  # if no output is set
+            arg_str = build_arg_string(kwargs)
+        elif "H" in kwargs.keys():  # if output is set
+            outfile = kwargs.pop("H")
+            if not outfile or not isinstance(outfile, str):
+                raise GMTInvalidInput("'output' should be a proper file name.")
+            arg_str = " ".join([build_arg_string(kwargs), f"-H > {outfile}"])
```

Comment:
Other functions in PyGMT which output to a file use `->` instead of `>`. I don't know why but maybe you can try if `->` also works.
```suggestion
            arg_str = " ".join([build_arg_string(kwargs), f"-H -> {outfile}"])
```

Reply:
I'm pretty sure it'll work, and I did notice it in other parts of the code but never quite understood why `->` is used. Is it meant to be a cross-platform thing?

Reply:
Ok, I was wrong, `->` doesn't work. The [`test_output_to_cpt_file`](https://github.com/GenericMappingTools/pygmt/blob/c22d0657d39a6aa4f7459a59f55fb45679d81ac4/pygmt/tests/test_makecpt.py#L77-L83) function fails with this error message:

```python-traceback
    def test_makecpt_output_to_cpt_file():
        """
        Save the generated static color palette table to a .cpt file
        """
        with GMTTempFile(suffix=".cpt") as cptfile:
>           makecpt(output=cptfile.name)

../pygmt/tests/test_makecpt.py:82: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../pygmt/helpers/decorators.py:187: in new_module
    return module_func(*args, **kwargs)
../pygmt/helpers/decorators.py:282: in new_module
    return module_func(*args, **kwargs)
../pygmt/mathops.py:65: in makecpt
    lib.call_module(module="makecpt", args=arg_str)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pygmt.clib.session.Session object at 0x7fe9ae6ad470>, module = 'makecpt', args = ' -H -> /tmp/pygmt-dz2kwkqs.cpt'

    def call_module(self, module, args):
        """
        Call a GMT module with the given arguments.
    
        Makes a call to ``GMT_Call_Module`` from the C API using mode
        ``GMT_MODULE_CMD`` (arguments passed as a single string).
    
        Most interactions with the C API are done through this function.
    
        Parameters
        ----------
        module : str
            Module name (``'pscoast'``, ``'psbasemap'``, etc).
        args : str
            String with the command line arguments that will be passed to the
            module (for example, ``'-R0/5/0/10 -JM'``).
    
        Raises
        ------
        GMTCLibError
            If the returned status code of the function is non-zero.
    
        """
        c_call_module = self.get_libgmt_func(
            "GMT_Call_Module",
            argtypes=[ctp.c_void_p, ctp.c_char_p, ctp.c_int, ctp.c_void_p],
            restype=ctp.c_int,
        )
    
        mode = self["GMT_MODULE_CMD"]
        status = c_call_module(
            self.session_pointer, module.encode(), mode, args.encode()
        )
        if status != 0:
            raise GMTCLibError(
                "Module '{}' failed with status code {}:\n{}".format(
>                   module, status, self._error_message
                )
            )
E           pygmt.exceptions.GMTCLibError: Module 'makecpt' failed with status code 71:
E           makecpt [ERROR]: Syntax error: No input files expected unless -E or -S are used

../pygmt/clib/session.py:490: GMTCLibError
--------------------------------------------------------------------------------- Captured stderr call ---------------------------------------------------------------------------------
makecpt [ERROR]: Syntax error: No input files expected unless -E or -S are used
```